bundles/admin/admin-layereditor/view/AdminLayerForm/VisualizationTabPane/RasterStyle.jsx

```diff
@@ -1,68 +1,86 @@
 import { Link } from './Link';
-import { StyledFormField, Border, DefaultStyle, StyleField, StyleSelect, InlineBlock } from './styled';
+import { StyledFormField, Border, InlineBlock } from './styled';
+import { RasterStyleSelect } from './RasterStyle/RasterStyleSelect';
+import { ServiceLegend } from './RasterStyle/ServiceLegend';
+
+const GLOBAL_LEGEND = 'legendImage';
+
+export const rasterStyleCapabilities = layer => {
+    const { options = {}, capabilities = {}, style } = layer;
+    const { styles = [] } = capabilities;
+    const { legends = {} } = options;
+    const selectedExists = styles.some(s => s.name === style);
+    const warnings = [];
+    if (style && !selectedExists) {
+        warnings.push('defaultStyle');
+    }
+    const additionals = legendsWithoutStyle(styles, legends);
+    const hasGlobal = additionals.find(name => name === GLOBAL_LEGEND);
+    // don't notify if global is only additional legend
+    if (additionals.length > 1 || (additionals.lenth === 1 && !hasGlobal)) {
+        warnings.push('additionalLegend');
+    }
+    if (styles.length > 0 && hasGlobal) {
+        warnings.push('globalWithStyles');
+    }
+    return warnings;
+};
+
+const legendsWithoutStyle = (styles, legends) => {
+    const styleNames = styles.map(s => s.name);
+    return Object.keys(legends)
+        .filter(name => !styleNames.includes(name));
+};
+
+const additionalLegendsToStyles = (styles, legends, globalTitle) => {
+    return legendsWithoutStyle(styles, legends)
+        .map(name => {
+            const style = {
+                name,
+                legend: legends[name]
+            };
+            const title = GLOBAL_LEGEND === name ? globalTitle : name;
+            style.title = title + ' ( ! )';
+            return style;
+        });
+};
 
 const RasterStyle = ({ layer, controller, getMessage }) => {
-    const [selected, setSelected] = useState(layer.style);
+    const { options = {}, capabilities = {}, style: defaultName } = layer;
+    const { styles = [] } = capabilities;
+    const { legends = {} } = options;
+    const additionalLegends = additionalLegendsToStyles(styles, legends, getMessage('styles.raster.legendImage'));
+    const styleOptions = [...styles, ...additionalLegends];
 
-    const isDefaultStyle = name => name === layer.style;
-    const getStyleLabel = style => {
-        const { name, title } = style;
-        const label = title || name;
-        if (isDefaultStyle(name)) {
-            return label + ' (' + getMessage('styles.default') + ')';
-        }
-        return label;
-    };
-    const onDefaultStyleChange = (styleName, selected) => {
-        const defaultStyle = selected ? styleName : '';
-        controller.setStyle(defaultStyle);
-    };
+    const firstOption = styleOptions.length > 0 ? styleOptions[0].name : '';
+    const [selected, setSelected] = useState(defaultName || firstOption);
+
+    const style = styles.find(s => s.name === selected);
```

Comment:
Shouldn't `layer.style` be used somewhere here?

Reply:
Nevermind, just spotted `defaultName` is destructured from `layer.style` at the start of the component.